src/javascript/eslint.ts

```diff
@@ -1,12 +1,60 @@
+import * as assert from "assert";
 import { Prettier } from "./prettier";
 import { Project, TaskStepOptions } from "..";
 import { DEFAULT_PROJEN_RC_JS_FILENAME } from "../common";
 import { Component } from "../component";
 import { NodeProject } from "../javascript";
+import { JavascriptFile, JavascriptRaw } from "../javascript-file";
 import { JsonFile } from "../json";
 import { Task } from "../task";
 import { YamlFile } from "../yaml";
 
+/**
+ * What format should the eslint file be.
+ *
+ */
+export enum EslintConfigFileFormat {
+  /**
+   * JavaScript file (new flat format) - using ESM-style imports/exports
+   *
+   * @see https://eslint.org/docs/latest/use/configure/configuration-files-new
+   */
+  JAVASCRIPT_FLAT_ESM = "flat-esm",
```

Comment:
I wonder if the format type is more important:

Enums should have the same string value as the enum:
```suggestion
  FLAT_JAVASCRIPT_ESM = "FLAT_JAVASCRIPT_ESM", 
```